Sources/SKCore/BuildSystem.swift

```diff
@@ -43,7 +40,10 @@ public protocol BuildSystem: AnyObject {
 
   /// Register the given file for build-system level change notifications, such
   /// as command line flag changes, dependency changes, etc.
-  func registerForChangeNotifications(for: URL)
+  ///
+  /// If the `BuildSystem` has cached build settings information, it is recommended to
+  /// immediately notify its delegate, as an optimization.
+  func registerForChangeNotifications(for: URL, language: Language)
```

Comment:
We should document that it's expected that there will be a notification with the initial settings if available.  The way it's currently worded makes it sound like an optimization, but this needs to always happen.  Also, something I hadn't considered: what do we do if there are no settings for this file? Maybe we could make the callback, but with `.removed`?

Reply:
Yep, there's also a question of whether the initial settings must be immediately called, in which case, why not have this method return them?

Reply:
Documented this in the latest commit